src/Temporalio/Worker/ReplaySafeLogger.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using Microsoft.Extensions.Logging;
+
+namespace Temporalio.Worker
+{
+    /// <summary>
+    /// Logger that will not log during replay.
+    /// </summary>
+    internal class ReplaySafeLogger : ILogger
```

Comment:
Wondering why you chose to put `Safe` in the name of this class, it's not about safety, just noise reduction.

Reply:
It's a common English approach for things like this. Could also be `ReplayAware`. What "X safe" would mean is that it's specifically built to be safe from the caveats that come with `X`, e.g. other English uses like "memory safe", "fail safe", etc.

Coincidentally durable tasks used similar naming. This name for us is only internal (this is not public API), but theirs is external: https://learn.microsoft.com/en-us/dotnet/api/microsoft.azure.webjobs.extensions.durabletask.durablecontextextensions.createreplaysafelogger?view=azure-dotnet.